cpp/include/cudf/aggregation.hpp

```diff
@@ -488,5 +490,39 @@ std::unique_ptr<Base> make_merge_sets_aggregation(null_equality nulls_equal = nu
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_merge_m2_aggregation();
 
+/**
+ * @brief Factory to create a TDIGEST aggregation
+ *
+ * Produces a tdigest (https://arxiv.org/pdf/1902.04023.pdf) column from input values.
+ * The input aggregation values are expected to be fixed-width numeric types.
+ *
+ * @param delta Parameter controlling compression level and accuracy on subsequent
+ * queries on the output tdigest data.  Delta places an upper bound on the size of
+ * the computed tdigests: A delta of 1000 will result in a tdigest containing no
+ * more than 1000 centroids (32 bytes each). Higher values of delta result in more
+ * accurate tdigest information.
+ *
+ * @returns A TDIGEST aggregation object.
+ */
+template <typename Base = aggregation>
+std::unique_ptr<Base> make_tdigest_aggregation(int delta = 1000);
```

Comment:
Where does 1000 come from?

Reply:
IIRC, the reference implementation uses `delta=100` for an "acceptable" error bounds, and we chose `1000` to see to check if that improves on the accuracy.

Reply:
Right, this is basically a magic number that defines the "compression" level of the resulting digest.  Although the larger the number is, the more precise the result is.   Internally, it puts a bound on the maximum number of centroid clusters a digest will have.  So with 1000, you'd end up with at most 1000 centroids (two doubles each).   As mithun points out, values of ~100 were commonly used in the source paper for millions of input bounds.

Since tdigest does not give any absolute guarantees on precision, I figured pumping into to a default of 1000 seemed reasonable, since that's a worst case resulting column size of ~32KB.